sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -0,0 +1,360 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   A pool of <see cref="TransportProducer" /> instances that automatically expire after a period of inactivity.
+    /// </summary>
+    ///
+    internal class TransportProducerPool : IAsyncDisposable
+    {
+        /// <summary>The period in minutes after which <see cref="PerformExpiration" /> is run.</summary>
+        private static readonly TimeSpan DefaultPerformExpirationPeriod = TimeSpan.FromMinutes(10);
+
+        /// <summary>
+        ///   The set of active Event Hub transport-specific producers specific to a given partition;
+        ///   intended to perform delegated operations.
+        /// </summary>
+        ///
+        private ConcurrentDictionary<string, PoolItem> Pool { get; }
+
+        /// <summary>
+        ///   An abstracted Event Hub transport-specific producer that is associated with the
```

Comment:
Side note:   This comment is super clear and helpful.

Reply:
Indeed.